build-support/bin/ci.sh

```diff
@@ -203,15 +201,21 @@ if [[ "${run_internal_backends:-false}" == "true" ]]; then
 fi
 
 if [[ "${run_python:-false}" == "true" ]]; then
-  if [[ "0/1" != "${python_unit_shard}" ]]; then
-    shard_desc=" [shard ${python_unit_shard}]"
-  fi
-  start_travis_section "CoreTests" "Running core python tests${shard_desc}"
+  start_travis_section "CoreTests" "Running core Python tests"
+  # TODO(#7772): Simplify below to always use V2 and drop the blacklist.
+  known_v2_failures_file="${REPO_ROOT}/build-support/unit_test_v2_blacklist.txt"
   (
-    ./pants.pex --tag='-integration' test.pytest --chroot \
-      "--test-pytest-test-shard=${python_unit_shard}" \
-      src/python:: tests/python:: -- "${PYTEST_PASSTHRU_ARGS[@]}"
-  ) || die "Core python test failure"
+    trap 'rm v2_targets.txt' EXIT
+    ./pants.pex --tag='-integration' --filter-type='python_tests' filter src/python:: tests/python:: > v2_targets.txt
+    v2_targets="$(comm -23 <(sort v2_targets.txt) <(sort "${known_v2_failures_file}"))"
+    # shellcheck disable=SC2086
```

Comment:
Does shellcheck accept post-scripts? Something like `# shellcheck disable=SC2086 [reason]` would be great.

Reply:
It does not :(

Shellcheck wants us to quote `$v2_targets`, but then this won't work because we are relying on bash to implicitly split the `\n`-delimited string for us.

Because this is temporary code and I could not find the fix within a few minutes, I was lazy in not fixing this. I think that this shows how to fix it? https://github.com/koalaman/shellcheck/wiki/SC2207. I tried implementing its suggestions and could not get it working.

(macOS does not pre-install `mapfile`, even when using a newer Bash, so we would have to use the `read -r` approach, which is finicky. This specific shellcheck is also what's blocking us from turning on shellcheck in CI! I can't figure out about 12 different instances of it currently.)

Reply:
OK. I think this would work:
```bash
v2_targets=($(comm -23 <(sort v2_targets.txt) <(sort "${known_v2_failures_file}")))
./pants.pex --no-v1 --v2 test.pytest "${v2_targets[@]}" -- "${PYTEST_PASSTHRU_ARGS[@]}"
```

Reply:
I don't understand github.

Reply:
Aha - the check actually explicitly NACKs my approach :/. OK.

Reply:
`--target-spec-file` allows for inputs from a file.